### PR TITLE
Reuse SignIn tokens if not expired

### PR DESCRIPTION
**Pull request by stroem**
The SignIn via secretsmanagerauth will now be cached until 5 minutes (const in code) the token expired

### CODE REVIEW

File: v2/auth/secretsmanagerauth/sign_in.go [hunk 1]
```diff
@@ -72,9 +74,32 @@ func SignIn(ctx context.Context) (err error) {
 		fetchingTokensMutex.Unlock()
 	}()
 
+	if tokens.AccessToken != "" {
```
Comment (stroem) [inline v2/auth/secretsmanagerauth/sign_in.go:78]: Maybe I should break this down to its own function?!

File: v2/auth/secretsmanagerauth/sign_in.go [hunk 2]
```diff
@@ -72,9 +74,32 @@ func SignIn(ctx context.Context) (err error) {
...
+		parser := jwt.Parser{
+			SkipClaimsValidation: true,
+		}
+
+		var claims jwt.StandardClaims
+		_, _, err := parser.ParseUnverified(tokens.AccessToken, &claims)
+		if err != nil {
```
Comment (stroem) [inline v2/auth/secretsmanagerauth/sign_in.go:85]: If error happends we should try to signin once again

File: v2/auth/secretsmanagerauth/sign_in.go [hunk 3]
```diff
@@ -72,9 +75,33 @@ func SignIn(ctx context.Context) (err error) {
 		fetchingTokensMutex.Unlock()
 	}()
 
+	if tokens.AccessToken != "" {
```
Comment (lonnblad) [inline v2/auth/secretsmanagerauth/sign_in.go:78]: maybe break out the expiresAt verification to its own function

**Comment by lonnblad** [comment]
the title of the PR could be a bit more descriptive of the actual work, like Added a a pre-check in SignIn for a valid cached token

File: v2/auth/secretsmanagerauth/sign_in.go [hunk 4]
```diff
@@ -44,6 +47,31 @@ func GetTokens() auth.Tokens {
...
+func isTokenValid(token string) bool {
+	if token != "" {
+		parser := jwt.Parser{
+			SkipClaimsValidation: true,
+		}
+
+		var claims jwt.StandardClaims
+		_, _, err := parser.ParseUnverified(token, &claims)
+
+		if err == nil {
```
Comment (lonnblad) [inline v2/auth/secretsmanagerauth/sign_in.go:60]: går att göra samma här, 
```
if err != nil {
  return false
}
```

File: v2/auth/secretsmanagerauth/sign_in.go [hunk 5]
```diff
@@ -44,6 +47,31 @@ func GetTokens() auth.Tokens {
...
+			// Verify if token still valid within the current time diff
+			// no need to sign in once again
+			ts := time.Now().Add(tokenExpireDurationDiff).Unix()
+			if claims.VerifyExpiresAt(ts, false) &&
```
Comment (lonnblad) [inline v2/auth/secretsmanagerauth/sign_in.go:64]: så kan den här bli
```
return claims.VerifyIssuedAt(ts, false) &&
  claims.VerifyIssuedAt(ts, false) &&
  claims.VerifyNotBefore(ts, false)
```

**Pull request merged**